src/core/lbgpu_cuda.cu

```diff
@@ -4265,4 +4268,65 @@ void lb_lbfluid_get_population( int xyz[3], float population_host[LBQ], int c )
+ * @param coupling              int denoting wether to use two- (0) or three-point (1) coupling for the velocity interpolation (Input)
+*/
+__global__ void lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel(LB_nodes_gpu n_a, CUDA_particle_data *particle_data, float* u_gpu, LB_node_force_gpu node_f, LB_rho_v_gpu *d_v, int coupling){
+  unsigned int index = blockIdx.y * gridDim.x * blockDim.x + blockDim.x * blockIdx.x + threadIdx.x;
+  if (index < para.number_of_particles) {
+    float position[3];
+    unsigned int node_index[8];
+    float mode[19*LB_COMPONENTS];
+    float delta[8];
+    position[0] = particle_data[index].p[0];
+    position[1] = particle_data[index].p[1];
+    position[2] = particle_data[index].p[2];
+    // Do the velocity interpolation
+    if (coupling == 0) {
+        interpolation_two_point_coupling(n_a, position, node_index, mode, nullptr, delta, &u_gpu[3*index]);
+    } else if (coupling == 1) {
+        interpolation_three_point_coupling(n_a, position, node_index, nullptr, delta, &u_gpu[3*index]);
+    }
+  }
+}
+
+/** interface to get fluid velocities at particle positions
+*/
+std::vector<float> lb_lbfluid_get_fluid_velocity_at_particle_positions(std::string coupling="twopoint") {
+  //call KERNEL and copy velocities from GPU to host
+  /** call of the particle kernel */
+  /** values for the particle kernel */
+  float* u_gpu;
+  cuda_safe_mem(cudaMalloc((void**)&u_gpu, 3*lbpar_gpu.number_of_particles*sizeof(float)));
+  int threads_per_block_particles = 64;
+  int blocks_per_grid_particles_y = 4;
+  int blocks_per_grid_particles_x = (lbpar_gpu.number_of_particles + threads_per_block_particles * blocks_per_grid_particles_y - 1) /
+                                    (threads_per_block_particles * blocks_per_grid_particles_y);
+  dim3 dim_grid_particles = make_uint3(blocks_per_grid_particles_x, blocks_per_grid_particles_y, 1);
+  if (coupling.compare("twopoint") == 0) {
+      KERNELCALL( lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel, dim_grid_particles, threads_per_block_particles,
+                  ( *current_nodes, gpu_get_particle_pointer(), u_gpu, node_f, device_rho_v, 0)
+                );
+  } else if (coupling.compare("threepoint") == 0) {
+      KERNELCALL( lb_lbfluid_get_fluid_velocity_at_particle_positions_kernel, dim_grid_particles, threads_per_block_particles,
+                  ( *current_nodes, gpu_get_particle_pointer(), u_gpu, node_f, device_rho_v, 1)
+                );
+  }
+
+  std::vector<float> u_host(3*lbpar_gpu.number_of_particles);
+  cuda_safe_mem(cudaMemcpy(u_host.data(), u_gpu, 3*lbpar_gpu.number_of_particles*sizeof(float), cudaMemcpyDeviceToHost));
+  cuda_safe_mem(cudaFree(u_gpu));
+  // Multiply all values of u_host by agrid/tau to get MD units.
+  std::transform(u_host.begin(), u_host.end(), u_host.begin(),
+                         std::bind1st(std::multiplies<float>(),lbpar_gpu.agrid/lbpar_gpu.tau));
```

Comment:
std::bind1st is deprecated in C++11 (and the use of std::bind is generally disencouraged.) This can always be replaced by a lambda like so:
```c++
std::transform(_, _, _, [&lbpar_gpu](float &f) { return f * lbpar_gpu.agrid/lbpar_gpu.tau });
```
Please do so.

Reply:
error: a variable with static storage duration cannot be captured in a lambda

Reply:
Ah sorry, you don't need to capture it because it's global. But then it should just work...

Reply:
Sorry, should be
```c++
  std::transform(u_host.begin(), u_host.end(), u_host.begin(),
                 [](float u) { return u * lbpar_gpu.agrid / lbpar_gpu.tau; });
```, the value has to be returned.